### ISSUE TITLE
Car Object Detection using DL

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/DL-Simplified#241]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Car Object Detection</br>
:red_circle: **Aim** : Create a DL model which will detect the objects(cars) from the given dataset.</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/dataclusterlabs/small-vehicle-images-far-vehicle-detection</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by NirmalKAhirwar** [comment]
please asigned this task to me. Car Object Detection using DL #241 .

FullName: Nirmal Kumar Ahirwar
Github Profile Link: https://github.com/NirmalKAhirwar
Email ID: [ahirwarnirmal2017@gmail.com](mailto:ahirwarnirmal2017@gmail.com)

Approach for this Project:
1] Perform exploratory data analysis on the dataset.
2] Apply data augmentation techniques to the dataset using the Albumentations Python library.
3] Create different deep learning models for Car (object detection), such as YOLOv8, Detectron, and CNN models like VGG, EfficientNet.
4] Compare the performance of all models using accuracy metrics and perform hyperparameter tuning to achieve higher accuracy on the test dataset.
5] Implement the models using TensorFlow or PyTorch.

What is your participant role? -contributor at Social Summer of Code (SSOC), Season-2

**Comment by abhisheks008** [comment]
According to the Code of Conduct and Contribution Guidelines, one contributor can be assigned one issue at a time.

**Comment by Devangini01** [comment]
please assign this task to me. Car Object Detection using DL 

FullName: Devangini Shinde
Github Profile Link: https://github.com/Devangini01
Email ID: shindedevangini04@gmail.com

Approach for this Project:
1] Perform exploratory data analysis on the dataset.
2] Apply data augmentation techniques to the dataset.
3] Create different deep learning models for Car (object detection), such as YOLOv8
4] Compare the performance of all models using accuracy metrics and perform hyperparameter tuning to achieve higher accuracy on the test dataset.
5] Implement the models using TensorFlow or PyTorch.

What is your participant role? -contributor at Social Summer of Code (SSOC), Season-2

**Comment by abhisheks008** [comment]
Issue assigned to you @Devangini01

**Comment by Krish-nobita903** [comment]
Please assign this task to me #241
FullName: Krish Srivastava
Github Profile Link: https://github.com/Krish-nobita903
Email ID: krish.rb.2517@gmail.com

Approach for this Project:
1)Preprocess the data set and perform the required analysis
2)Choose tensorflow based pre defined model like YOLO , R-CNN etc
3)Fine-tune the selected pre-trained model on your car detection dataset
4)Augment training dataset to increase its size and diversity. 
5)Train the model on labeled dataset and evaluate the model on testing dataset to assess its performance. 

What is  your participant role? : Contributor at SWOC(Social Winter of Code)

**Comment by abhisheks008** [comment]
Let the program start officially. Issues will be assigned after that. Till then go through the repository as well as the `README` file.

**Comment by Krish-nobita903** [comment]
ok thank you sir

**Comment by abhisheks008** [comment]
Hi @Krish-nobita903 do want to contribute in this issue with the approach you have given earlier?

**Comment by Krish-nobita903** [comment]
Yes bhaiya

**Comment by abhisheks008** [comment]
Issue assigned to you @Krish-nobita903. You can start working on it. All the best!

**Comment by Krish-nobita903** [comment]
thanks bhaiya

**Comment by binguliki** [comment]
@abhisheks008 Hey Bro 👋, I have worked on this task earlier in my learning period , I would like to add my knowledge of YOLO algorithm and faster R-CNN inorder to identify cars , later this could be used a base for detecting vehicles in some different task 

Full name : Bingumalla Likith
GitHub Profile Link : [binguliki](https://github.com/binguliki)
Email ID : blikith86@gmail.com
Participant ID (if applicable): NA
Approach for this Project : I would be using pretrained YOLO algorithm and Faster RCNN as they are capable of detecting objects quickly , For metrics I used be using Accuracy along with Precision , Recall and F1 score
What is your participant role? `GSSoC'24 | Contributor`


Can you please Assign me this under GSSoC ' 24 with appropriate Level tag , You can have a bet on me I would surely complete the project and PR you soon

**Comment by abhisheks008** [comment]
I like your enthusiam but one issue at a time @binguliki

**Comment by binguliki** [comment]
@abhisheks008  Hey bro , I'll take this one , later i would go for the other one , currently i'm not assigned any issue.

**Comment by YashSachan2** [comment]
Full name : Yash Sachan
GitHub Profile Link : https://github.com/YashSachan2
Email ID: [yash.sachan.ece22@itbhu.ac.in](mailto:yash.sachan.ece22@itbhu.ac.in)
Approach: I will preprocess the images for the size suitable for the pretrained models and then fine tune and analyze the performance of pre trained models like resnet,darknet,mobilenetV2,YOLOv5,YOLOv8.Metrics for comparison  would be recall and precision. 
What is your participant role?
GSSoc'24
Please assign me this issue @abhisheks008

**Comment by abhisheks008** [comment]
I like your enthusiasm but one issue at a time @YashSachan2

**Comment by kyra-09** [comment]
hey , @abhisheks008 , I would like to work on this issue and I'll be using KerasCV or Pytorch for localization and detection of different types of vehicles listed using transfer learning approach

**Comment by Rithish5513U** [comment]
Full name : Rithish S
GitHub Profile Link : https://github.com/Rithish5513U
Email ID : rithish.satish@gmail.com
Approach for this Project :
I will use data augmentation techniques to preprocess images. I will use vgg16, mobilenet and yolo for the given problem. 
I have a query, I only found vehicles other than car in the given dataset. Can u explain me about what should i do with this dataset or can i use other datasets for this project?
What is your participant role? (Mention the Open Source program)
Gssoc'24 Contributor

**Comment by umairullah0905** [comment]
Hi, sir @abhisheks008 , I listed my approach down, please go through it and assign me under tag GSSOC'24

Full name : Umair Ullah
GitHub Profile Link : https://github.com/umairullah0905
Email ID : umairulla05@gmail.com
Participant ID (if applicable):
Approach for this Project :
Algorithms - CNN architecture model such VGG16, RESNET50 can be classify the images.
Preprocessing - Image resizing, normalization, encoding and augmentation if required
Model Comparison and Selection - Evaluate Performance of all models based on the metrics obtained and will Choose the model that shows the best balance between accuracy, generalizability, and computational efficiency
What is your participant role? GSSOC'24

**Comment by abhisheks008** [comment]
> Full name : Rithish S GitHub Profile Link : https://github.com/Rithish5513U Email ID : [rithish.satish@gmail.com](mailto:rithish.satish@gmail.com) Approach for this Project : I will use data augmentation techniques to preprocess images. I will use vgg16, mobilenet and yolo for the given problem. I have a query, I only found vehicles other than car in the given dataset. Can u explain me about what should i do with this dataset or can i use other datasets for this project? What is your participant role? (Mention the Open Source program) Gssoc'24 Contributor

Can you find a compatible dataset for the same?

**Comment by Rithish5513U** [comment]
@abhisheks008 Check this dataset : https://www.kaggle.com/datasets/sshikamaru/car-object-detection
This dataset contains images taken from a video frame by frame. Can I use this dataset to detect cars or can I search for any other??

**Comment by abhisheks008** [comment]
> @abhisheks008 Check this dataset : https://www.kaggle.com/datasets/sshikamaru/car-object-detection This dataset contains images taken from a video frame by frame. Can I use this dataset to detect cars or can I search for any other??

You can go ahead with this dataset.

Assigned @Rithish5513U
